src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -903,6 +937,29 @@ internal void SuppressFastKillInHandleProcessExit()
                 ExecuteFastKillInProcessExit = false;
             }
 
+            internal IDictionary<GrainId, IGrainInfo> GetDirectory()
+            {
+                return silo.localGrainDirectory.DirectoryPartition.GetItems();
+            }
+          
+            internal IDictionary<GrainId, IGrainInfo> GetDirectoryForTypenamesContaining(string expr)
+            {
+                var x = new Dictionary<GrainId, IGrainInfo>();
+                foreach (var kvp in GetDirectory())
+                {
+                    if (kvp.Key.IsSystemTarget || kvp.Key.IsClient || !kvp.Key.IsGrain)
+                        continue;// Skip system grains, system targets and clients
+                    if (silo.catalog.GetGrainTypeName(kvp.Key).Contains(expr))
+                        x.Add(kvp.Key, kvp.Value);
+                }
+                return x;
+            }
+
+            internal void InjectMultiClusterConfiguration(MultiClusterConfiguration config)
```

Comment:
I could not find any usages of this either, nor of `GetDirectory()` (outside of GetDirectoryForTypenamesContaining)